File: solution/LC169_Majority_Element.java
package solution;

public class LC169_Majority_Element {
	public int majorityElement(int[] nums) {
        int major = nums[0];
        int count = 0;
        for(int i:nums){
            if(count == 0){
                major = i;
                count++;
            } else if(i == major){
                count ++;
            } else {
                count --;
            }
        }
        return major;
    }
}

File: javascript/solution/LC169_Majority_Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    var res = nums[0];
    var count = 1;
    for(let i = 1; i < nums.length; i++){
        if(nums[i] == res){
            count++;
        } else if(count == 0){
            res = nums[i];
            count = 1;
        } else {
            count--;
        }
    }
    return res;
};
